### ISSUE TITLE
hydroxide@v0.2.15 fails to build (/protonmail/contacts.go, /protonmail/crypto.go and srp.go)

### ISSUE THREAD

**Issue by adamsteen** [emersion/hydroxide#106]
```
$ uname -a
OpenBSD x220.adamsteen.com.au 6.7 GENERIC.MP#26 amd64
```
```
$ go version
go version go1.13.9 openbsd/amd64
```
```
$ GO111MODULE=on go get github.com/emersion/hydroxide/cmd/hydroxide
go: finding github.com/emersion/hydroxide v0.2.15
go: downloading github.com/emersion/hydroxide v0.2.15
go: extracting github.com/emersion/hydroxide v0.2.15
go: downloading golang.org/x/crypto v0.0.0-20200221231518-2aa609cf4a9d
go: downloading github.com/howeyc/gopass v0.0.0-20190910152052-7cb4b85ec19c
go: downloading github.com/emersion/go-imap-specialuse v0.0.0-20161227184202-ba031ced6a62
go: downloading github.com/emersion/go-message v0.11.1
go: downloading github.com/emersion/go-imap v1.0.4
go: extracting github.com/howeyc/gopass v0.0.0-20190910152052-7cb4b85ec19c
go: downloading github.com/emersion/go-bcrypt v0.0.0-20170822072041-6e724a1baa63
go: extracting github.com/emersion/go-imap-specialuse v0.0.0-20161227184202-ba031ced6a62
go: downloading github.com/emersion/go-smtp v0.12.1
go: extracting github.com/emersion/go-message v0.11.1
go: downloading github.com/emersion/go-imap-move v0.0.0-20190710073258-6e5a51a5b342
go: extracting github.com/emersion/go-imap v1.0.4
go: downloading github.com/emersion/go-textwrapper v0.0.0-20160606182133-d0e65e56babe
go: downloading github.com/emersion/go-sasl v0.0.0-20191210011802-430746ea8b9b
go: extracting github.com/emersion/go-imap-move v0.0.0-20190710073258-6e5a51a5b342
go: downloading golang.org/x/text v0.3.2
go: extracting github.com/emersion/go-smtp v0.12.1
go: extracting github.com/emersion/go-bcrypt v0.0.0-20170822072041-6e724a1baa63
go: extracting github.com/emersion/go-sasl v0.0.0-20191210011802-430746ea8b9b
go: extracting github.com/emersion/go-textwrapper v0.0.0-20160606182133-d0e65e56babe
go: downloading github.com/boltdb/bolt v1.3.1
go: extracting github.com/boltdb/bolt v1.3.1
go: extracting golang.org/x/crypto v0.0.0-20200221231518-2aa609cf4a9d
go: downloading github.com/emersion/go-webdav v0.2.1-0.20200203205455-3ea3818dd842
go: downloading golang.org/x/sys v0.0.0-20200223170610-d5e6a3e2c0ae
go: downloading github.com/emersion/go-vcard v0.0.0-20191221110513-5f81fa0d3cc7
go: downloading github.com/martinlindhe/base36 v1.0.0
go: extracting github.com/martinlindhe/base36 v1.0.0
go: extracting github.com/emersion/go-vcard v0.0.0-20191221110513-5f81fa0d3cc7
go: extracting github.com/emersion/go-webdav v0.2.1-0.20200203205455-3ea3818dd842
go: extracting golang.org/x/sys v0.0.0-20200223170610-d5e6a3e2c0ae
go: extracting golang.org/x/text v0.3.2
go: finding github.com/emersion/go-imap-move v0.0.0-20190710073258-6e5a51a5b342
go: finding github.com/emersion/go-imap-specialuse v0.0.0-20161227184202-ba031ced6a62
go: finding github.com/emersion/go-imap v1.0.4
go: finding github.com/emersion/go-smtp v0.12.1
go: finding github.com/emersion/go-vcard v0.0.0-20191221110513-5f81fa0d3cc7
go: finding github.com/emersion/go-message v0.11.1
go: finding github.com/emersion/go-webdav v0.2.1-0.20200203205455-3ea3818dd842
go: finding golang.org/x/text v0.3.2
go: finding github.com/emersion/go-sasl v0.0.0-20191210011802-430746ea8b9b
go: finding github.com/howeyc/gopass v0.0.0-20190910152052-7cb4b85ec19c
go: finding golang.org/x/crypto v0.0.0-20200221231518-2aa609cf4a9d
go: finding github.com/emersion/go-bcrypt v0.0.0-20170822072041-6e724a1baa63
go: finding golang.org/x/sys v0.0.0-20200223170610-d5e6a3e2c0ae
go: finding github.com/martinlindhe/base36 v1.0.0
go: finding github.com/emersion/go-textwrapper v0.0.0-20160606182133-d0e65e56babe
go: finding github.com/boltdb/bolt v1.3.1
# github.com/emersion/hydroxide/protonmail
go/pkg/mod/github.com/emersion/hydroxide@v0.2.15/protonmail/contacts.go:167:66: too many arguments in call to openpgp.CheckArmoredDetachedSignature
        have (openpgp.KeyRing, *bytes.Buffer, io.Reader, nil)
        want (openpgp.KeyRing, io.Reader, io.Reader)
go/pkg/mod/github.com/emersion/hydroxide@v0.2.15/protonmail/contacts.go:193:55: too many arguments in call to openpgp.CheckArmoredDetachedSignature
        have (openpgp.KeyRing, *strings.Reader, *strings.Reader, nil)
        want (openpgp.KeyRing, io.Reader, io.Reader)
go/pkg/mod/github.com/emersion/hydroxide@v0.2.15/protonmail/crypto.go:39:21: subkey.PublicKey.KeyExpired undefined (type *packet.PublicKey has no field or method KeyExpired)
go/pkg/mod/github.com/emersion/hydroxide@v0.2.15/protonmail/crypto.go:56:137: i.SelfSignature.SigExpired undefined (type *packet.Signature has no field or method SigExpired)
go/pkg/mod/github.com/emersion/hydroxide@v0.2.15/protonmail/crypto.go:73:21: subkey.PublicKey.KeyExpired undefined (type *packet.PublicKey has no field or method KeyExpired)
go/pkg/mod/github.com/emersion/hydroxide@v0.2.15/protonmail/crypto.go:87:80: i.SelfSignature.SigExpired undefined (type *packet.Signature has no field or method SigExpired)
go/pkg/mod/github.com/emersion/hydroxide@v0.2.15/protonmail/srp.go:45:41: too many arguments in call to openpgp.CheckDetachedSignature
        have (openpgp.EntityList, *bytes.Reader, io.Reader, nil)
        want (openpgp.KeyRing, io.Reader, io.Reader)
```

I am not sure what else you need, sorry I don't know much about `go` to help here.

**Comment by emersion** [comment]
Hmm. Looks like this is a Go issue: https://github.com/golang/go/issues/30354#issuecomment-466479708

Cloning the repo and running `go build ./cmd/hydroxide` should work.

**Comment by dvalter** [comment]
I believe it's the same as #105 
TL;DR
> I just didn't notice that golang.org/x/crypto was replaced with github.com/ProtonMail/crypto

**Comment by adamsteen** [comment]
@emersion I am now up and running, just to testing with my newer keys.

Should the instructions change on the readme now?

**Comment by emersion** [comment]
Yeah, it would be nice to update the README instructions. Can you send a pull request?

**Comment by adamsteen** [comment]
@emersion I will see what i can do.

**Comment by harleylang** [comment]
Thank you for updating the docs!!

As a side note for docker users, the documented installation code works but pruning is required beforehand.

**Comment by emersion** [comment]
Closing because the README has been updated.
